From 20eb37b08782fa0c811b491d232fcf831f80ba6d Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:28:36 -0400
Subject: [pci] aer: Support ACPI HEST AER error sources for PCI domains other than 0

Message-id: <20140719042836.19438.83410.stgit@gir.stowe>
Patchwork-id: 86018
O-Subject: [RHEL7.1 PATCH 56/81] PCI/AER: Support ACPI HEST AER error sources for PCI domains other than 0
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: 339c0fc58618b717565f109d68da50c5bc1a0510

commit 339c0fc58618b717565f109d68da50c5bc1a0510
Author: Betty Dall <betty.dall@hp.com>
Date:   Thu Dec 5 08:08:24 2013 -0700

    PCI/AER: Support ACPI HEST AER error sources for PCI domains other than 0

    In the discussion for this set of patches [link below], Bjorn Helgaas
    pointed out that the ACPI HEST AER error sources do not have the PCIe
    segment number associated with the bus.  I worked with the ACPI spec and
    got this change to definition of the "Bus" field into the recently released
    ACPI Spec 5.0a section 18.3.2.3-5:

      Identifies the PCI Bus and Segment of the device.  The Bus is encoded in
      bits 0-7.  For systems that expose multiple PCI segment groups, the
      segment number is encoded in bits 8-23 and bits 24-31 must be zero.  For
      systems that do not expose multiple PCI segment groups, bits 8-31 must be
      zero.  If the GLOBAL flag is specified, this field is ignored.

    This patch makes use of the new definition in the only place in the kernel
    that uses the acpi_hest_aer_common's bus field.

    This depends on 36f3615152c1 ("ACPICA: Add helper macros to extract
    bus/segment numbers from HEST table.")

    Link: http://lkml.kernel.org/r/1370542251-27387-1-git-send-email-betty.dall@hp.com
    Signed-off-by: Betty Dall <betty.dall@hp.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/pcie/aer/aerdrv_acpi.c b/drivers/pci/pcie/aer/aerdrv_acpi.c
index 4d69917..0190657 100644
--- a/drivers/pci/pcie/aer/aerdrv_acpi.c
+++ b/drivers/pci/pcie/aer/aerdrv_acpi.c
@@ -23,10 +23,10 @@
 static inline int hest_match_pci(struct acpi_hest_aer_common *p,
      struct pci_dev *pci)
 {
- return (0           == pci_domain_nr(pci->bus) &&
-   p->bus      == pci->bus->number &&
-   p->device   == PCI_SLOT(pci->devfn) &&
-   p->function == PCI_FUNC(pci->devfn));
+ return   ACPI_HEST_SEGMENT(p->bus) == pci_domain_nr(pci->bus) &&
+   ACPI_HEST_BUS(p->bus)     == pci->bus->number &&
+   p->device                 == PCI_SLOT(pci->devfn) &&
+   p->function               == PCI_FUNC(pci->devfn);
 }
 
 static inline bool hest_match_type(struct acpi_hest_header *hest_hdr,
-- 
1.7.1